src/App.tsx

```diff
@@ -1,7 +1,43 @@
-import './App.scss';
+import { useContext, useEffect } from 'react';
+import { Header } from './components/base/Header/Header.component';
+import { DispatchContext, StatesContext } from './store/GlobalStateProvider';
+import { Outlet } from 'react-router-dom';
+import { MenuPage } from './pages/Menu/Menu.page';
+import cn from 'classnames';
+import { Footer } from './components/base/Footer/Footer.component';
+import { getCategories } from './api/products';
 
-export const App = () => (
-  <div className="App">
-    <h1>Product Catalog</h1>
-  </div>
-);
+export const App = () => {
+  const dispatch = useContext(DispatchContext);
+  const { isMenuOpen, cart } = useContext(StatesContext);
+
+  useEffect(() => {
+    dispatch({
+      type: 'updateTotalCartItems',
+      payload: cart.reduce((acc, prod) => acc + (prod.quantity ?? 1), 0),
+    });
+  }, [cart]);
+
+  useEffect(() => {
+    Promise.allSettled([
+      getCategories()
+        .then(cats => dispatch({ type: 'loadCategories', payload: cats }))
+        .catch(() => 'something wrong while fetching products'),
+    ]).finally(() => dispatch({ type: 'isReady', payload: true }));
+  }, []);
+
+  return (
+    <div className="App">
+      <h1 className="h1Hidden">Product Catalog</h1>
+      <header className="header">
+        <Header />
```

Comment:
i think it makes more sense for the <header> tag to be inside the header properly